base/special/trig.jl

```diff
@@ -860,8 +860,8 @@ function cospi(x::T) where T<:Union{Integer,Rational}
     end
 end
 
-sinpi(x::Integer) = x >= 0 ? zero(float(x)) : -zero(float(x))
-cospi(x::Integer) = isodd(x) ? -one(float(x)) : one(float(x))
+sinpi(x::Integer) = 0
```

Comment:
Should the return type match the input, i.e. `zero(x)`?

Reply:
See my other comment. I think for consistency, the return type of `sinpi` should always be the same as `cospi`, though, even if we could actually return `Unsigned` here.

Reply:
I think it's fine if they differ, so that some computations remain `Unsigned` if all their underlying functions allow it. If someone plugs results of `sinpi` and `cospi` together, type promotion will fix that difference.